src/target/riscv/riscv.c

```diff
@@ -1874,6 +1874,12 @@ static int riscv_halt_go_all_harts(struct target *target)
 		return ERROR_FAIL;
 	if (state == RISCV_STATE_HALTED) {
 		LOG_TARGET_DEBUG(target, "Hart is already halted.");
+		if (target->state != TARGET_HALTED) {
+			target->state = TARGET_HALTED;
+			enum riscv_halt_reason halt_reason = riscv_halt_reason(target);
+			if (set_debug_reason(target, halt_reason) != ERROR_OK)
+				return ERROR_FAIL;
```

Comment:
`return ERROR_FAIL`. 

I'm not sure that return of ERROR_FAIL is the right way. Logically we should just set halt_reason to unknown/undefined and report LOG_TARGET_WARNING/LOG_TARGET_ERROR

Reply:
It may be the case that there is a misunderstanding. I probably selected a wrong line of code. Consider this:

1. We want to halt all relevant targets.  This is the primary goal.
2. For a particular hart - we already know that it is halted. So our primary goal is complete.
3. We also want to figure out debug reason. We do this by calling `riscv013_halt_reason`. This procedure can fail.
4. I don't think that we need to abort the whole process in this case. We can just notify user that we could not determine the reason for the halt.